### PR TITLE
feat(Divider): add divider component

### LINKED ISSUE #2575: Create Divider Feature
This got merged in Core: https://github.com/patternfly/patternfly-next/pull/2080

Create a divider component with the same examples as in Core.

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #2575

### CODE REVIEW

File: packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export const Divider: React.FunctionComponent<DividerProps> = ({
+  className = '',
+  'role': role = '',
+  variant = 'hr',
+  ...props
+}: DividerProps) => {
+  const Component: any = variant;
+
+  return (
+    <Component {...props} className={css(styles.divider, className)} role={role}/>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx]: Can you spread props at the end here so they take precedence.

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DividerDemo/DividerDemo.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    window.scrollTo(0, 0);
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <Divider/>
+
+        <ul>
+          <li>List item one</li>
```
Comment (tlabaj) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/DividerDemo/DividerDemo.tsx]: after you update the component you can update these to remove the `role`

File: yarn.lock [hunk 3]
```diff
@@ -2503,6 +2503,7 @@
 "@patternfly/patternfly@2.33.8":
   version "2.33.8"
   resolved "https://registry.yarnpkg.com/@patternfly/patternfly/-/patternfly-2.33.8.tgz#2885847b4c49d8606abed17a0415da24fec222aa"
+  integrity sha512-tkT1GrdIyvGiDvNHATOa7XLMLKdztpra8ngS8qhCcEWBKh9MWNj5ysjOrkna32bSF9GEkR4TFX492W8ESTHWvQ==
```
Comment (tlabaj) [inline yarn.lock]: Can you revert the changes in this file.  It should not have been updated with the changes you made.

File: packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  div = 'div'
+}
+
+export interface DividerProps extends React.HTMLProps<HTMLElement> {
+  /** additional classes added to the Badge */
+  className?: string;
+  /** Adds a role to the divider for accessibility */
+  'role'?: string;
+  /** the component type to use */
+  variant?: 'hr' | 'li' | 'div' ;
```
[the remark this answers is not in the document]
Comment (christiemolloy) [reply packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx]: could you explain what you mean here :)

File: packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+export const Divider: React.FunctionComponent<DividerProps> = ({
+  className = '',
+  variant = 'hr',
+  ...props
+}: DividerProps) => {
+  const Component: any = variant;
+
+  return (
+    <Component className={css(styles.divider, className)} role={ variant!='hr' ? 'separator' : undefined } {...props}/>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx]: I think you want to do something like this instead. 

```suggestion
    <Component className={css(styles.divider, className)} {...(variant != 'hr' && {role : 'seperator' })}  {...props}/>
```

File: packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (christiemolloy) [reply packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx]: thanks updated

File: packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx (hunk 5 above)
Comment (christiemolloy) [reply packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx]: That makes sense thank you @tlabaj !

File: packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+stage: 'early'
+---
+import { Divider } from '@patternfly/react-core/dist/esm/experimental';
+
+## Divider using hr (default)
+```js
+import React from 'react';
+import { Divider } from '@patternfly/react-core/dist/esm/experimental';
+
+DividerHr = (
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.md]: Did you mean for this to be a function? e.g.
```suggestion
DividerHr = () => (
```
same comment for other examples.

File: packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import { cpus } from 'os';
+
+export enum DividerVariant {
+  hr = 'hr',
+  li = 'li',
+  div = 'div'
+}
+
+export interface DividerProps extends React.HTMLProps<HTMLElement> {
+  /** additional classes added to the Badge */
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx]: Can you change the prop descriptions to Sentence case.

File: packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+export const Divider: React.FunctionComponent<DividerProps> = ({
+  className = '',
+  component = DividerVariant.hr,
+  ...props
+}: DividerProps) => {
+  const Component: any = component;
+
+  return (
+    <Component className={css(styles.divider, className)} {...(component != 'hr' && {role : 'seperator' })} {...props} />
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/experimental/components/Divider/Divider.tsx]:
```suggestion
    <Component className={css(styles.divider, className)} {...(component != 'hr' && {role : 'separator' })} {...props} />
```

File: packages/patternfly-4/react-core/src/experimental/components/Divider/__snapshots__/Divider.test.tsx.snap [hunk 9]
```diff
@@ -0,0 +1,21 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`divider using div 1`] = `
+<div
+  className="pf-c-divider"
+  role="seperator"
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/experimental/components/Divider/__snapshots__/Divider.test.tsx.snap]:
```suggestion
  role="separator"
```

File: packages/patternfly-4/react-core/src/experimental/components/Divider/__snapshots__/Divider.test.tsx.snap [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+exports[`divider using hr 1`] = `
+<hr
+  className="pf-c-divider"
+/>
+`;
+
+exports[`divider using li 1`] = `
+<li
+  className="pf-c-divider"
+  role="seperator"
```
Comment (mcoker) [inline packages/patternfly-4/react-core/src/experimental/components/Divider/__snapshots__/Divider.test.tsx.snap]:
```suggestion
  role="separator"
```
